### PR TITLE
Clean up and reorder render layer define directives.

### PR DESCRIPTION
**Pull request by betstick**
Reorder ``#define`` directives to be ordered from oldest to newest render layers: ``USE_DX9`` -> ``USE_DX11`` -> ``USE_OGL``
Not all could be organized this way without substantial code duplication.

Got rid of almost if not all of the negative(?) defines for render layers. Examples:
`` #ifndef USE_OGL`` -> ``#if defined(USE_DX9) || defined(USE_DX11)``
``#if !defined(USE_DX9) && !defined(USE_DX11)`` -> ``#if defined(USE_OGL)``
This should help in the event that a new render layer is added to prevent unexpected code to be compiled/run. This predicates on there only ever being one layer active at a time per "unit" or whatever. So ``#if defined(USE_DX9) && defined(USE_DX11)`` should always be false.

Explicitly defined some conditions for readability and to prevent unpredictable behavior in the event that the render layers are changed. Example:
```
#ifdef USE_DX9
    //some code
#else
    //some code
#endif
``` 
becomes the following:
```
#ifdef USE_DX9
    //some code
#elif defined(USE_DX11) || defined(USE_OGL)
    //some code
#else
#error No graphics API selected or in use!
#endif
``` 

Finally, added some fail conditions so that if no render layers are used for a given code fragment the compiler will error out. Example:
```
#if defined(USE_DX9) || defined(USE_DX11)
    //some code
#elif defined(USE_OGL)
    //some code
#else
#error No graphics API selected or in use!
#endif
```

I tried to avoid any changes possible to code style/format outside of reordering lines. 

Note, I named the commit a bad name and I can't figure out how to undo this. Also, the created/modified dates on the files are totally screwed up since I tar'd and untar'd a local copy of the repo a few times across Windows and Linux to test compilation. This is not a 100% cleanup, but it was everything I could find across several text searches of the repo for specific directives. This is my first pull request so please let me know if I messed up/broke something.

### CODE REVIEW

File: src/Layers/xrRender/Blender_Recorder_R2.cpp [hunk 1]
```diff
@@ -137,8 +137,10 @@ u32 CBlender_Compile::r_Sampler(
     {
 #if defined(USE_DX11)
         r_dx10Texture(_name, texture, true);
-#else
+#elif defined(USE_DX9) || defined(USE_OGL)
         i_Texture(dwStage, texture);
+#else
+#error No graphics API selected or enabled!
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/Blender_Recorder_R2.cpp]: Nested preprocessor directives should have an indent:
```suggestion
#   error No graphics API selected or enabled!
```
This applies to all `#error`s in this PR.

File: src/Layers/xrRender_R2/r2_rendertarget_accum_reflected.cpp [hunk 2]
```diff
@@ -159,9 +160,12 @@ void CRenderTarget::accum_reflected(light* L)
                     draw_volume(L);
                 }
                 StateManager.SetSampleMask(0xffffffff);
-#   endif // USE_OGL
+#   elif defined(USE_OGL)
+                VERIFY(!"Only optimized MSAA is supported in OpenGL");
+
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender_R2/r2_rendertarget_accum_reflected.cpp]:
```suggestion
```

File: src/Layers/xrRender_R2/r2_rendertarget_accum_reflected.cpp [hunk 3]
```diff
@@ -113,7 +113,10 @@ void CRenderTarget::accum_reflected(light* L)
...
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender_R2/r2_rendertarget_accum_reflected.cpp]:
```suggestion
```

File: src/Layers/xrRender_R2/r2.cpp [hunk 4]
```diff
@@ -535,7 +547,7 @@ void CRender::create()
     o.dx10_minmax_sm = ps_r3_minmax_sm;
     o.dx10_minmax_sm_screenarea_threshold = 1600 * 1200;
 
-#ifndef USE_OGL
+#if defined(USE_DX9) || defined(USE_DX11)
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender_R2/r2.cpp]: This is for DX11 only.. 
```suggestion
#if defined(USE_DX11)
```

File: src/Layers/xrRender/stats_manager.h [hunk 5]
```diff
@@ -51,7 +51,7 @@ class stats_manager
 
 u32 get_format_pixel_size(D3DFORMAT format);
 
-#ifndef USE_DX9
+#if defined(USE_DX11) || defined(USE_OGL)
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/stats_manager.h]: This is DX11-only
```suggestion
#if defined(USE_DX11)
```

File: src/Layers/xrRender/stats_manager.cpp [hunk 6]
```diff
@@ -258,4 +270,4 @@ u32 get_format_pixel_size(DXGI_FORMAT format)
         // Do not consider extraordinary formats.
         return 0;
 }
-#endif
+#endif
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/stats_manager.cpp]: Don't remove whitespace in the end of file
```suggestion
#endif

```

File: src/Layers/xrRender/r_constants_cache.h [hunk 7]
```diff
@@ -182,4 +184,4 @@ class	ECORE_API  R_constants
     }
 };
 */
-#endif
+#endif
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/r_constants_cache.h]:
```suggestion
#endif

```

**Comment by betstick** [comment]
I've added the suggested changes from this site except for the indentation which I had to push separately from my editor to do it all at once. Apologies for the mistakes in the original push. I'll try to be more careful with the code formatting in the future. Also, I think I need to go back and manually check out a few hundred instances where I may have made this mistake in a less easy to regex manner. It is also something I will be bringing up in the Discord for discussion.

File: src/Layers/xrRender/Blender_Recorder_R2.cpp [hunk 8]
```diff
@@ -137,8 +137,10 @@ u32 CBlender_Compile::r_Sampler(
     {
 #if defined(USE_DX11)
         r_dx10Texture(_name, texture, true);
-#else
+#elif defined(USE_DX9) || defined(USE_OGL)
         i_Texture(dwStage, texture);
+#else
+#	error No graphics API selected or enabled!
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/Blender_Recorder_R2.cpp]: Please, always use spaces, not tabs.
```suggestion
#   error No graphics API selected or enabled!
```

**Comment by Xottab-DUTY** [comment]
There is a problem with this PR: it changes file mode from 100644 to 100755

**Comment by Xottab-DUTY** [comment]
There is a problem: PR changes file mode from 100644 to 100755 for some files

**Comment by betstick** [comment]
I've fixed the file permissions issues and made changes in my system to prevent this from happening again. I also fixed the tabs issues relevant to the render layer directives I'm already touching. Fingers crossed I didn't break something new.

File: src/Layers/xrRender_R2/r2.cpp [hunk 9]
```diff
@@ -134,18 +136,18 @@ static class cl_sun_shafts_intensity : public R_constant_setup
...
         // TODO: OGL: Implement AlphaRef.
-#   ifndef USE_OGL
+#   if defined(USE_DX11)
         StateManager.BindAlphaRef(C);
-#   endif
+#   endif // !USE_OGL
     }
 } binder_alpha_ref;
-#endif
+#endif // !USE_DX9
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender_R2/r2.cpp]:
```suggestion
#endif
```

File: src/Layers/xrRender_R2/r2.cpp [hunk 10]
```diff
@@ -134,18 +136,18 @@ static class cl_sun_shafts_intensity : public R_constant_setup
...
 static class cl_alpha_ref : public R_constant_setup
 {
     virtual void setup(R_constant* C)
     {
         // TODO: OGL: Implement AlphaRef.
-#   ifndef USE_OGL
+#   if defined(USE_DX11)
         StateManager.BindAlphaRef(C);
-#   endif
+#   endif // !USE_OGL
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender_R2/r2.cpp]:
```suggestion
#   endif
```

File: src/Layers/xrRender_R2/r2.cpp [hunk 11]
```diff
@@ -220,7 +222,7 @@ void CRender::create()
     Device.seqFrame.Add(this, REG_PRIORITY_HIGH + 0x12345678);
 
     m_skinning = -1;
-#ifndef USE_DX9
+#if defined(USE_DX11) || defined(USE_OGL)
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender_R2/r2.cpp]: Better to leave it as is for now.
```suggestion
#ifndef USE_DX9
```

File: src/Layers/xrRender_R2/r2.cpp [hunk 12]
```diff
@@ -382,17 +390,19 @@ void CRender::create()
...
     if (o.nvdbt)
         Msg("* NV-DBT supported and used");
 
-#if !defined(USE_OGL)
+#if defined(USE_DX9) || defined(USE_DX11)
     o.no_ram_textures = (strstr(Core.Params, "-noramtex")) ? TRUE : ps_r__common_flags.test(RFLAG_NO_RAM_TEXTURES);
     if (o.no_ram_textures)
         Msg("* Managed textures disabled");
-#endif
+#endif // !USE_OGL
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender_R2/r2.cpp]:
```suggestion
#endif
```

File: src/Layers/xrRender/r__sector_traversal.cpp [hunk 13]
```diff
@@ -135,7 +135,7 @@ void CPortalTraverser::dbg_draw()
     RCache.set_xform_world(Fidentity);
     RCache.set_xform_view(Fidentity);
     RCache.set_xform_project(Fidentity);
-#ifndef USE_DX9
+#if defined(USE_DX11) || defined(USE_OGL)
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/r__sector_traversal.cpp]:
```suggestion
#if defined(USE_DX11) || defined(USE_OGL) // when we don't have FFP support
```

File: src/Layers/xrRender/r__sector.cpp [hunk 14]
```diff
@@ -74,7 +74,7 @@ void CPortal::OnRender()
             Device.SetNearer(TRUE);
       
         RCache.set_Shader(RImplementation.m_WireShader);
-#ifndef USE_DX9
+#if defined(USE_DX11) || defined(USE_OGL)
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/r__sector.cpp]:
```suggestion
#if defined(USE_DX11) || defined(USE_OGL) // when we don't have FFP support
```

File: src/Layers/xrRender/r__sector.cpp [hunk 15]
```diff
@@ -56,7 +56,7 @@ void CPortal::OnRender()
         RCache.set_xform_world(Fidentity);
         // draw solid
         RCache.set_Shader(RImplementation.m_SelectionShader);
-#ifndef USE_DX9
+#if defined(USE_DX11) || defined(USE_OGL)
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/r__sector.cpp]:
```suggestion
#if defined(USE_DX11) || defined(USE_OGL) // when we don't have FFP support
```

File: src/Layers/xrRender/r__sync_point.cpp [hunk 16]
```diff
@@ -82,5 +82,6 @@ void R_sync_point::End()
     q_sync_count = (q_sync_count + 1) % HW.Caps.iGPUNum;
     CHK_DX(EndQuery((ID3DQuery*)q_sync_point[q_sync_count]));
 }
-
-#endif // USE_OGL
+#else
+#    error No graphics API selected or enabled!
+#endif // USE_OGL
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/r__sync_point.cpp]:
```suggestion
#endif

```

File: src/Layers/xrRender/xrRender_console.cpp [hunk 17]
```diff
@@ -22,12 +22,11 @@ const xr_token qsmapsize_token[] =
     { "3072", 3072 },
     { "3584", 3584 },
     { "4096", 4096 },
-#ifndef USE_DX9
+#if defined(USE_DX11) || defined(USE_OGL)  // XXX: check if this really supported on OpenGL
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/xrRender_console.cpp]:
```suggestion
#if defined(USE_DX11) || defined(USE_OGL) // XXX: check if values more than 8192 are supported on OpenGL
```

File: src/Layers/xrRender/D3DXRenderBase.h [hunk 18]
```diff
@@ -56,7 +56,7 @@ class D3DXRenderBase : public IRender, public pureFrame
     xr_vector<R_dsgraph::mapNormalTextures::value_type *> nrmTexturesTemp;
 
     xr_vector<R_dsgraph::mapMatrixVS::value_type *> matVS;
-#ifndef USE_DX9
+#if defined(USE_DX11) || defined(USE_OGL)
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/D3DXRenderBase.h]: Many modern graphics APIs have geometry shaders, so, let's better to leave it as is:
```suggestion
#ifndef USE_DX9
```

File: src/Layers/xrRender/D3DXRenderBase.h [hunk 19]
```diff
@@ -46,7 +46,7 @@ class D3DXRenderBase : public IRender, public pureFrame
 
     // Runtime structures
     xr_vector<R_dsgraph::mapNormalVS::value_type *> nrmVS;
-#ifndef USE_DX9
+#if defined(USE_DX11) || defined(USE_OGL) //USE_DX9
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/D3DXRenderBase.h]: Many modern graphics APIs have geometry shaders, so, let's better to leave it as is:
```suggestion
#ifndef USE_DX9
```

File: src/Layers/xrRender/ColorMapManager.cpp [hunk 20]
```diff
@@ -41,22 +44,26 @@ void ColorMapManager::UpdateTexture(const shared_str& strTexName, int iTex)
...
     else
     {
-#ifdef USE_OGL
-        m_CMap[iTex]->surface_set(GL_TEXTURE_2D, 0);
-#else
+#if defined(USE_DX9) || defined(USE_DX11)
         m_CMap[iTex]->surface_set(nullptr);
-#endif // USE_OGL
+#elif defined(USE_OGL)
+m_CMap[iTex]->surface_set(GL_TEXTURE_2D, 0);
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/ColorMapManager.cpp]:
```suggestion
        m_CMap[iTex]->surface_set(GL_TEXTURE_2D, 0);
```

File: src/Layers/xrRender/QueryHelper.h [hunk 21]
```diff
@@ -89,23 +72,44 @@ IC HRESULT ReleaseQuery(ID3DQuery* pQuery)
...
+    CHK_GL(glDeleteQueries(1, &query));
+    return S_OK;
+}
+
+#else
+#    error No graphics API selected or enabled!
 #endif
 
-#endif // QueryHelper_included
+#endif // QueryHelper_included
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/QueryHelper.h]:
```suggestion
#endif // QueryHelper_included

```

File: src/Layers/xrRender_R2/r2_rendertarget_phase_PP.cpp [hunk 22]
```diff
@@ -146,25 +146,27 @@ void CRenderTarget::phase_pp()
...
+    pv->set(du + 0, dv + float(_h), p_color, p_gray, r0.x, r1.y, l0.x, l1.y, n0.x, n1.y);
     pv++;
     pv->set(du + float(_w), dv + 0, p_color, p_gray, r1.x, r0.y, l1.x, l0.y, n1.x, n0.y);
     pv++;
-#endif // USE_OGL
+    pv->set(du + float(_w), dv + float(_h), p_color, p_gray, r1.x, r1.y, l1.x, l1.y, n1.x, n1.y);
+    pv++;
+#else
+#    error No graphics API selected or enabled!
+#endif // !USE_OGL
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender_R2/r2_rendertarget_phase_PP.cpp]:
```suggestion
#endif // USE_DX9 || USE_DX11
```

File: src/Layers/xrRender_R2/r2_rendertarget_phase_smap_S.cpp [hunk 23]
```diff
@@ -62,25 +62,27 @@ void CRenderTarget::phase_smap_spot_tsh(light* L)
...
+        pv->set(EPS, float(_h + EPS), d_Z, d_W, C, p0.x, p1.y);
         pv++;
         pv->set(float(_w + EPS), EPS, d_Z, d_W, C, p1.x, p0.y);
         pv++;
-#endif // USE_OGL
+        pv->set(float(_w + EPS), float(_h + EPS), d_Z, d_W, C, p1.x, p1.y);
+        pv++;
+#else
+#    error No graphics API selected or enabled!
+#endif // !USE_OGL
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender_R2/r2_rendertarget_phase_smap_S.cpp]:
```suggestion
#endif // USE_DX9 || USE_DX11
```

File: src/Layers/xrRender_R2/r2.cpp [hunk 24]
```diff
@@ -477,24 +489,18 @@ void CRender::create()
...
-#ifndef USE_DX9
-#   ifdef USE_OGL
-    o.dx10_sm4_1 = true;
-#   else
+#if defined(USE_DX11) || defined(USE_OGL)
+#   if defined(USE_DX11)
     o.dx10_sm4_1 = ps_r2_ls_flags.test((u32)R3FLAG_USE_DX10_1);
     o.dx10_sm4_1 = o.dx10_sm4_1 && (HW.FeatureLevel >= D3D_FEATURE_LEVEL_10_1);
+#   elif defined(USE_OGL)
+    o.dx10_sm4_1 = true;
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender_R2/r2.cpp:499]:
```suggestion
    o.dx10_sm4_1 = true;
#else
#   error No graphics API selected or enabled!
```

File: src/Layers/xrRender_R2/r2.cpp [hunk 25]
```diff
@@ -505,6 +511,12 @@ void CRender::create()
     // o.dx10_msaa_hybrid	= ps_r2_ls_flags.test(R3FLAG_MSAA_HYBRID);
     o.dx10_msaa_hybrid = ps_r2_ls_flags.test((u32)R3FLAG_USE_DX10_1);
     o.dx10_msaa_hybrid &= !o.dx10_msaa_opt && o.dx10_msaa && (HW.FeatureLevel >= D3D_FEATURE_LEVEL_10_1);
+#   elif defined(USE_OGL)
+    // TODO: OGL: temporary disabled, need to fix it
+    o.dx10_msaa = false;
+    o.dx10_msaa_samples = 0;
+    o.dx10_msaa_opt = o.dx10_msaa;
+    o.dx10_msaa_hybrid = false;
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender_R2/r2.cpp:521]:
```suggestion
    o.dx10_msaa_hybrid = false;
#else
#   error No graphics API selected or enabled!
```

File: src/Layers/xrRender_R2/r2.cpp [hunk 26]
```diff
@@ -477,24 +489,18 @@ void CRender::create()
             o.hbao_vectorized = true;
         o.ssao_opt_data = true;
     }
-#endif
+#endif // !USE_DX9
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender_R2/r2.cpp]:
```suggestion
#endif // USE_DX9
```

File: src/Layers/xrRender_R2/r2.cpp [hunk 27]
```diff
@@ -424,14 +434,14 @@ void CRender::create()
...
+#   if defined(USE_DX11)
+    o.volumetricfog = ps_r2_ls_flags.test(R3FLAG_VOLUMETRIC_SMOKE);
+#   elif defined(USE_OGL)
     // TODO: OGL: temporary disabled, need to fix it
     o.volumetricfog = false;
-#   else
-    o.volumetricfog = ps_r2_ls_flags.test(R3FLAG_VOLUMETRIC_SMOKE);
 #   endif
-#endif
+#endif // !USE_DX9
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender_R2/r2.cpp]:
```suggestion
#endif
```

File: src/Layers/xrRender/HOM.cpp [hunk 28]
```diff
@@ -446,7 +448,7 @@ void CHOM::OnRender()
             // draw solid
             Device.SetNearer(TRUE);
             RCache.set_Shader(RImplementation.m_SelectionShader);
-#ifndef USE_DX9
+#if defined(USE_DX11) || defined(USE_OGL)
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/HOM.cpp]:
```suggestion
#if defined(USE_DX11) || defined(USE_OGL) // when we don't have FFP support
```

File: src/Layers/xrRender/HOM.cpp [hunk 29]
```diff
@@ -462,7 +464,7 @@ void CHOM::OnRender()
                 Device.SetNearer(TRUE);
             }
             RCache.set_Shader(RImplementation.m_SelectionShader);
-#ifndef USE_DX9
+#if defined(USE_DX11) || defined(USE_OGL)
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/HOM.cpp]:
```suggestion
#if defined(USE_DX11) || defined(USE_OGL) // when we don't have FFP support
```

File: src/Layers/xrRender/ParticleEffect.cpp [hunk 30]
```diff
@@ -32,34 +32,36 @@ static void ApplyTexgen(const Fmatrix& mVP)
 {
     Fmatrix mTexgen;
 
-#ifdef USE_OGL
+#if defined(USE_DX9)// USE_DX9
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/ParticleEffect.cpp]:
```suggestion
#if defined(USE_DX9)
```

File: src/Layers/xrRender/r_constants.cpp [hunk 31]
```diff
@@ -218,13 +218,12 @@ void R_constant_table::merge(R_constant_table* T)
...
-#ifndef USE_DX9
-            C->gs = src->gs;
-#ifdef USE_DX11
+#if defined(USE_DX11)
             C->hs = src->hs;
             C->ds = src->ds;
             C->cs = src->cs;
-#endif
+#elif defined(USE_OGL)
+            C->gs = src->gs;
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/r_constants.cpp]:
```suggestion
```

File: src/Layers/xrRender/r_constants.cpp [hunk 32]
```diff
@@ -218,13 +218,12 @@ void R_constant_table::merge(R_constant_table* T)
             C->type = src->type;
             C->ps = src->ps;
             C->vs = src->vs;
-#ifndef USE_DX9
-            C->gs = src->gs;
-#ifdef USE_DX11
+#if defined(USE_DX11)
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/r_constants.cpp]:
```suggestion
#ifndef USE_DX9
            C->gs = src->gs;
#   if defined(USE_DX11)
```

File: src/Layers/xrRender/r_constants.h [hunk 33]
```diff
@@ -74,21 +74,25 @@ struct ECORE_API R_constant_load
...
     {
-#ifdef USE_OGL
+#if defined(USE_DX9) || defined(USE_DX11)
+        return (index == C.index) && (cls == C.cls);
+#elif defined(USE_OGL)
         return (index == C.index) && (cls == C.cls) && (location == C.location) && (program == C.program);
 #else
-        return (index == C.index) && (cls == C.cls);
+#    error No graphics API selected or enabled!
 #endif // USE_OGL
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/r_constants.h]:
```suggestion
#endif
```

File: src/Layers/xrRender/r_constants.h [hunk 34]
```diff
@@ -101,14 +105,17 @@ struct ECORE_API R_constant : public xr_resource
...
+
+#if defined(USE_DX11) || defined(USE_OGL)
+	R_constant_load gs; //ogl OR dx11
+#endif
+
+#if defined(USE_DX11)
     R_constant_load hs;
     R_constant_load ds;
     R_constant_load cs;
 #endif
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/r_constants.h]: - ps, vs and others should be near each other, they are related. Thus, whitespace is not needed.
- GS (Geometry Shaders) exist on many graphics APIs, so "if not exist" check is shorter and we won't need to add each new API to this check.
```suggestion
#ifndef USE_DX9
    R_constant_load gs;
#   if defined(USE_DX11)
    R_constant_load hs;
    R_constant_load ds;
    R_constant_load cs;
#   endif
#endif
```

File: src/Layers/xrRender/stats_manager.cpp [hunk 35]
```diff
@@ -44,13 +44,15 @@ void stats_manager::increment_stats_rtarget(ID3DTexture2D* buff)
...
     _D3DPOOL pool = D3DPOOL_MANAGED;
-#ifndef USE_DX9
-    D3D_TEXTURE2D_DESC desc;
-    buff->GetDesc(&desc);
-#else
+#if defined(USE_DX9)
     D3DSURFACE_DESC desc;
     buff->GetLevelDesc(0, &desc);
     pool = desc.Pool;
+#elif defined(USE_DX11) || defined(USE_OGL)
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/stats_manager.cpp]:
```suggestion
#elif defined(USE_DX11)
```

File: src/Layers/xrRender/stats_manager.cpp [hunk 36]
```diff
@@ -62,14 +64,16 @@ void stats_manager::increment_stats_vb(ID3DVertexBuffer* buff)
     if (buff == nullptr || GEnv.isDedicatedServer)
         return;
 
-#ifndef USE_DX9
+#if defined(USE_DX9)
+    D3DVERTEXBUFFER_DESC desc;
+    buff->GetDesc(&desc);
+    increment_stats(desc.Size, enum_stats_buffer_type_vertex, desc.Pool, buff);
+#elif defined(USE_DX11) || defined(USE_OGL)
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/stats_manager.cpp]:
```suggestion
#elif defined(USE_DX11)
```

File: src/Layers/xrRender/stats_manager.cpp [hunk 37]
```diff
@@ -78,14 +82,16 @@ void stats_manager::increment_stats_ib(ID3DIndexBuffer* buff)
     if (buff == nullptr || GEnv.isDedicatedServer)
         return;
 
-#ifndef USE_DX9
+#if defined(USE_DX9)
+    D3DINDEXBUFFER_DESC desc;
+    buff->GetDesc(&desc);
+    increment_stats(desc.Size, enum_stats_buffer_type_index, desc.Pool, buff);
+#elif defined(USE_DX11) || defined(USE_OGL)
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/stats_manager.cpp]:
```suggestion
#elif defined(USE_DX11)
```

File: src/Layers/xrRender/stats_manager.cpp [hunk 38]
```diff
@@ -100,13 +106,15 @@ void stats_manager::decrement_stats_rtarget(ID3DTexture2D* buff)
...
     _D3DPOOL pool = D3DPOOL_MANAGED;
-#ifndef USE_DX9
-    D3D_TEXTURE2D_DESC desc;
-    buff->GetDesc(&desc);
-#else
+#if defined(USE_DX9)
     D3DSURFACE_DESC desc;
     buff->GetLevelDesc(0, &desc);
     pool = desc.Pool;
+#elif defined(USE_DX11) || defined(USE_OGL)
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/stats_manager.cpp]:
```suggestion
#elif defined(USE_DX11)
```

File: src/Layers/xrRender/stats_manager.cpp [hunk 39]
```diff
@@ -123,14 +131,16 @@ void stats_manager::decrement_stats_vb(ID3DVertexBuffer* buff)
     if ((refcnt = buff->Release()) > 1)
         return;
 
-#ifndef USE_DX9
+#if defined(USE_DX9)
+    D3DVERTEXBUFFER_DESC desc;
+    buff->GetDesc(&desc);
+    decrement_stats(desc.Size, enum_stats_buffer_type_vertex, desc.Pool, buff);
+#elif defined(USE_DX11) || defined(USE_OGL)
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/stats_manager.cpp]:
```suggestion
#elif defined(USE_DX11)
```

File: src/Layers/xrRender/stats_manager.cpp [hunk 40]
```diff
@@ -144,14 +154,16 @@ void stats_manager::decrement_stats_ib(ID3DIndexBuffer* buff)
     if ((refcnt = buff->Release()) > 1)
         return;
 
-#ifndef USE_DX9
+#if defined(USE_DX9)
+    D3DINDEXBUFFER_DESC desc;
+    buff->GetDesc(&desc);
+    decrement_stats(desc.Size, enum_stats_buffer_type_index, desc.Pool, buff);
+#elif defined(USE_DX11) || defined(USE_OGL)
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/stats_manager.cpp]:
```suggestion
#elif defined(USE_DX11)
```

File: src/Layers/xrRender/stats_manager.cpp [hunk 41]
```diff
@@ -239,7 +251,7 @@ u32 get_format_pixel_size(D3DFORMAT format)
     }
 }
 
-#ifndef USE_DX9
+#if defined(USE_DX11) || defined(USE_OGL)
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/stats_manager.cpp]:
```suggestion
#if defined(USE_DX11)
```

File: src/Layers/xrRender/r_constants.cpp [hunk 42]
```diff
@@ -220,11 +220,10 @@ void R_constant_table::merge(R_constant_table* T)
             C->vs = src->vs;
 #ifndef USE_DX9
             C->gs = src->gs;
-#ifdef USE_DX11
+#   if defined(USE_DX11)
             C->hs = src->hs;
             C->ds = src->ds;
             C->cs = src->cs;
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/r_constants.cpp:226]:
```suggestion
            C->cs = src->cs;
#   endif
```

File: src/Layers/xrRender/dxFontRender.cpp [hunk 43]
```diff
@@ -101,7 +101,7 @@ void dxFontRender::OnRender(CGameFont& owner)
                     clr2 = color_rgba(_R, _G, _B, _A);
                 }
 
-#ifndef USE_DX9 // Vertex shader will cancel a DX9 correction, so make fake offset
+#if defined(USE_DX11) || defined(USE_OGL) // Vertex shader will cancel a DX9 correction, so make fake offset
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/dxFontRender.cpp]: Special case for DX9 only.
```suggestion
#ifndef USE_DX9 // Vertex shader will cancel a DX9 correction, so make fake offset
```

File: src/Layers/xrRender/dxEnvironmentRender.cpp [hunk 44]
```diff
@@ -332,7 +336,7 @@ void dxEnvironmentRender::RenderClouds(CEnvironment& env)
     RCache.set_xform_world(mXFORM);
     RCache.set_Geometry(clouds_geom);
     RCache.set_Shader(clouds_sh);
-#ifndef USE_OGL
+#if defined(USE_DX9) || defined(USE_DX11)
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/dxEnvironmentRender.cpp]:
```suggestion
#if defined(USE_DX9) || defined(USE_DX11) // XXX: why it this disabled for OGL?
```

File: src/Layers/xrRender/dxDebugRender.cpp [hunk 45]
```diff
@@ -18,7 +18,7 @@ void dxDebugRender::Render()
         return;
 
     RCache.set_xform_world(Fidentity);
-#ifndef USE_DX9
+#if defined(USE_DX11) || defined(USE_OGL)
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/dxDebugRender.cpp]:
```suggestion
#if defined(USE_DX11) || defined(USE_OGL) // when we don't have FFP support
```

**Review by Xottab-DUTY — APPROVED** [review]

**Comment by Xottab-DUTY** [comment]
Thank you!

I reverted a small number of changes. Please, let me know, if you still need a refactored version.

**Pull request merged**